datalad/tests/utils.py

```diff
@@ -231,6 +235,14 @@ def newfunc(*arg, **kw):
                 # well -- if something wrong just proceed with defaults
                 pass
 
+        # if DATALAD_TESTS_TEMPDIR is set, use that as directory,
+        # let mktemp handle it otherwise. However, an explicitly provided
+        # dir=... will override this.
+        directory = os.environ.get('DATALAD_TESTS_TEMPDIR')
```

Comment:
ideally any new functionality should be tested... for mocking environ, see https://github.com/datalad/datalad/blob/HEAD/datalad/tests/test_log.py#L49